hazelcast/src/main/java/com/hazelcast/nio/Bits.java

```diff
@@ -337,4 +348,34 @@ public static long combineToLong(int x, int y) {
     public static int extractInt(long value, boolean lowerBits) {
         return (int) ((lowerBits) ? value : (value >> 32));
     }
+
+    /**
+     * Fast method of finding the next power of 2 greater than or equal to the supplied value.
+     * <p/>
+     * If the value is &lt;= 0 then 1 will be returned.
+     * <p/>
+     * This method is not suitable for {@link Integer#MIN_VALUE} or numbers greater than 2^30.
+     *
+     * @param value from which to search for next power of 2
+     * @return The next power of 2 or the value itself if it is a power of 2
+     */
+    public static int findNextPositivePowerOfTwo(final int value) {
```

Comment:
Indeed, I was just writing the same. My point was intended to apply to `(1L << 62) + 1`, which is a positive value.

However, both methods actually do the same in this respect and `findNextPositivePowerOfTwo` documents it is not suitable for arguments above `1 << 30`.

What "positive" refers to, and where these methods differ, is the behavior against nonpositive arguments. `findNextPositivePowerOfTwo` returns 1 and `nextPowerOfTwo` returns 0. Here `nextPowerOfTwo` actually breaks its contract by returning 0 for 0.